src/tools/rbd_mirror/image_replayer/journal/ReplayStatusFormatter.cc

```diff
@@ -242,7 +253,28 @@ void ReplayStatusFormatter<I>::format(std::string *description) {
   root_obj["non_primary_position"] = to_json_object(m_mirror_position);
   root_obj["entries_behind_primary"] = (
     m_entries_behind_master > 0 ? m_entries_behind_master : 0);
-  *description = json_spirit::write(root_obj);
+
+  m_bytes_per_second(0);
+  root_obj["bytes_per_second"] = round_to_two_places(
+    m_bytes_per_second.get_average());
+
+  m_entries_per_second(0);
+  auto entries_per_second = m_entries_per_second.get_average();
+  root_obj["entries_per_second"] = round_to_two_places(entries_per_second);
+
+  if (m_entries_behind_master > 0 && entries_per_second > 0) {
+    auto seconds_until_synced = round_to_two_places(
+      m_entries_behind_master / entries_per_second);
```

Comment:
Not sure if we want to improve this, but I think it will be underestimate if the primary image is actively writing at that time. If we had something like `new_journal_entries_per_second`, for the case  `new_journal_entries_per_second << entries_per_second` the formula could be
```
seconds_until_synced = entries_behind_master / entries_per_second * (1 + new_journal_entries_per_second / entries_per_second)
```
For larger `new_journal_entries_per_second` it would need to include additional terms, and if `new_journal_entries_per_second >= entries_per_second` it would just mean that the sync never complete.

Alternatively I think we could estimate `seconds_until_synced` by measuring the rate of `entries_behind_master` change.  If it is growing -- the sync will never complete. If it is decreasing, `seconds_until_synced = entries_behind_master / rate`.

Reply:
I thought about that and had put something in initially to track it, but the issue is that it's going to be a highly inaccurate stat since we only discover new master entries every 30 seconds or so. Since we don't get the actual timestamp from the master position commit, we just have to guess that it's X entries per 30 seconds.